internal/cmd/skupper/site/site_create.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright © 2024 Skupper Team <skupper@googlegroups.com>
+*/
+package site
+
+import (
+	"context"
+	"fmt"
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/client"
+	"github.com/skupperproject/skupper/internal/cmd/skupper/utils"
+	"github.com/skupperproject/skupper/pkg/apis/skupper/v1alpha1"
+	skupperv1alpha1 "github.com/skupperproject/skupper/pkg/generated/client/clientset/versioned/typed/skupper/v1alpha1"
+	"github.com/skupperproject/skupper/pkg/site"
+	"github.com/skupperproject/skupper/pkg/utils/validator"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	siteCreateLong = `A site is a place where components of your application are running. 
+Sites are linked to form application networks.
+There can be only one site definition per namespace.`
+
+	linkAccessTypes = []string{"route", "loadbalancer", "nodeport", "nginx-ingress-v1", "contour-http-proxy", "ingress"}
+)
+
+type CreateFlags struct {
+	enableLinkAccess bool
+	linkAccessType   string
+	linkAccessHost   string
+	serviceAccount   string
+}
+
+type CmdSiteCreate struct {
+	Client             skupperv1alpha1.SkupperV1alpha1Interface
+	KubeClient         kubernetes.Interface
+	CobraCmd           cobra.Command
+	flags              CreateFlags
+	options            map[string]string
+	siteName           string
+	serviceAccountName string
+	Namespace          string
+}
+
+func NewCmdSiteCreate() *CmdSiteCreate {
+
+	options := make(map[string]string)
+	skupperCmd := CmdSiteCreate{options: options, flags: CreateFlags{}}
+
+	cmd := cobra.Command{
+		Use:    "create <name>",
+		Short:  "Create a new site",
+		Long:   siteCreateLong,
+		PreRun: skupperCmd.NewClient,
+		Run: func(cmd *cobra.Command, args []string) {
+			utils.HandleErrorList(skupperCmd.ValidateInput(args))
+			skupperCmd.InputToOptions()
+			utils.HandleError(skupperCmd.Run())
+		},
+		PostRunE: func(cmd *cobra.Command, args []string) error {
+			return skupperCmd.WaitUntilReady()
+		},
+	}
+
+	skupperCmd.CobraCmd = cmd
+	skupperCmd.AddFlags()
+
+	return &skupperCmd
+}
+
+func (cmd *CmdSiteCreate) NewClient(cobraCommand *cobra.Command, args []string) {
+	cli, err := client.NewClient("", "", "")
+	utils.HandleError(err)
+
+	cmd.Client = cli.SkupperClient.SkupperV1alpha1()
+	cmd.KubeClient = cli.KubeClient
+	cmd.Namespace = cli.Namespace
+}
+
+func (cmd *CmdSiteCreate) AddFlags() {
+	cmd.CobraCmd.Flags().BoolVar(&cmd.flags.enableLinkAccess, "enable-link-access", false, "Enable external access for links from remote sites")
```

Comment:
In the latest CRD on v2, the site spec has a single 'linkAccess' field. This can take the values 'none', 'default', 'loadbalancer', 'route' etc. None implies no link access. Default implies link access using the default type for the controller (which if not explicitly configured will be route if available, loadbalancer if not at present).

Reply:
My bad! fixed: 562f1b9be9a1d060d709e6e726e61b89ee0d9f57